pkg/scheduler/core/generic_scheduler.go

```diff
@@ -687,22 +689,22 @@ func PrioritizeNodes(
 		}
 	})
 
-	for i, priorityConfig := range priorityConfigs {
-		if priorityConfig.Reduce == nil {
+	for i := range priorityConfigs {
+		if priorityConfigs[i].Reduce == nil {
```

Comment:
Just out of curiosity. There will never be a situation when Function != nil and Reduce != nil, right?

Reply:
That's right. The Map/Reduce functions are the new style for priority functions. The old-style was using "Function". No priority function is supposed to mix the two and none of our priority functions do that.